src/main/java/com/bookstore/bookstoreapi/data/service/CustomerService

```diff
@@ -0,0 +1,52 @@
+package com.bookstore.bookstoreapi.service;
+
+import com.bookstore.bookstoreapi.data.entity.Customer;
+import com.bookstore.bookstoreapi.data.repository.CustomerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CustomerService {
+
+    private final CustomerRepository customerRepository;
+
+    @Autowired
+    public CustomerService(CustomerRepository customerRepository) {
+        this.customerRepository = customerRepository;
+    }
+
+    public List<Customer> getAllCustomers() {
+        return customerRepository.findAll();
+    }
+
+    public Optional<Customer> getCustomerById(Long customerId) {
+        return customerRepository.findById(customerId);
+    }
+
+    public Customer createCustomer(Customer customer) {
+        return customerRepository.save(customer);
+    }
+
+    public Customer updateCustomer(Long customerId, Customer updatedCustomer) {
+        Optional<Customer> existingCustomer = customerRepository.findById(customerId);
+        if (existingCustomer.isPresent()) {
+            Customer customerToUpdate = existingCustomer.get();
+            // Update the customer fields as needed
+            customerToUpdate.setFirstName(updatedCustomer.getFirstName());
+            customerToUpdate.setLastName(updatedCustomer.getLastName());
+            customerToUpdate.setEmail(updatedCustomer.getEmail());
+            customerToUpdate.setAddress(updatedCustomer.getAddress());
+            return customerRepository.save(customerToUpdate);
+        } else {
+            // Customer not found
+            return null;
+        }
+    }
+
+    public void deleteCustomer(Long customerId) {
```

Comment:
There's no need to do all these void methods, the methods you need already exist, so you in this case you need to delete these and use the ones provided by the `JPARepository`